### PR TITLE
New Data Source: `azurerm_backup_protected_file_share`

### LINKED ISSUE #13216: Support for azurerm_backup_protected_file_share data source
Hello,

I'd like to request a data source for azurerm_backup_protected_file_share.

Reason being is that the policy id is needed whenever one wants to protect a file share and a data source come in handy in cases where the recovery vault already exists.

Best regards
Mario

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by SudoSpartanDan**
## Summary

Adding new Data Source `azurerm_backup_protected_file_share`.

Fixes #13216

### CODE REVIEW

**Review by favoretti — CHANGES_REQUESTED** [review]
Hi there and thank you for this PR. Left a few minor comments/suggestions inline. If we can get this patched up - this otherwise LGTM.

File: internal/services/recoveryservices/backup_protected_file_share_data_source.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return fmt.Errorf("making Read request on Recovery Service Protected File Share %q (Vault %q Resource Group %q): %+v", id.Name, id.VaultName, id.ResourceGroup, err)
+	}
+
+	d.SetId(id.ID())
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("recovery_vault_name", id.VaultName)
+
+	if properties := resp.Properties; properties != nil {
+		if item, ok := properties.AsAzureFileshareProtectedItem(); ok {
+			sourceResourceID := strings.Replace(*item.SourceResourceID, "Microsoft.storage", "Microsoft.Storage", 1) // The SDK is returning inconsistent capitalization
```
[the remark this answers is not in the document]
Comment (SudoSpartanDan) [reply internal/services/recoveryservices/backup_protected_file_share_data_source.go]: Good catch, updated

File: website/docs/d/backup_protected_file_share.html.markdown [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+subcategory: "Recovery Services"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_backup_protected_file_share"
+description: |-
+  Gets information about an existing Azure Backup Protected File Share.
+---
+
+# Data Source: azurerm_backup_protected_file_share
+
+Use this data source to access information about an existing VM Backup Policy.
```
[the remark this answers is not in the document]
Comment (SudoSpartanDan) [reply website/docs/d/backup_protected_file_share.html.markdown]: Good catch, updated

**Review by katbyte — CHANGES_REQUESTED** [review]
Have a test failure here:
```
------- Stdout: -------
=== RUN   TestAccDataSourceBackupProtectedFileShare_basic
=== PAUSE TestAccDataSourceBackupProtectedFileShare_basic
=== CONT  TestAccDataSourceBackupProtectedFileShare_basic
    testcase.go:110: Step 1/1 error: Error running apply: exit status 1
        
        Error: making Read request on Recovery Service Protected File Share "acctest-ss-220215235951921355" (Vault "acctest-VAULT-220215235951921355" Resource Group "acctestRG-backup-220215235951921355"): backup.ProtectedItemsClient#Get: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BMSUserErrorProtectedItemNameIncorrectFormat" Message="Protected Item name is not in the correct format."
        
          with data.azurerm_backup_protected_file_share.test,
          on terraform_plugin_test.tf line 71, in data "azurerm_backup_protected_file_share" "test":
          71: data "azurerm_backup_protected_file_share" "test" {
        
--- FAIL: TestAccDataSourceBackupProtectedFileShare_basic (707.64s)
FAIL
```

**Comment by SudoSpartanDan** [comment]
I am not able to get the tests working properly, looks like the other associated tests with this resource have been failing as well. I'm going to close this until those other acceptance tests are able to work properly.

**Pull request closed without merging**